lib/date_range.rb

```diff
@@ -0,0 +1,56 @@
+require 'date'
+
+module Hotel
+  class DateRange
+    attr_reader :check_in, :check_out
+    
+    def initialize(check_in, check_out)
+      @check_in = check_in
+      @check_out = check_out
+      
+      validate_date_range
+      validate_date_object
+    end
+    
+    
+    def validate_date_range
+      if @check_out < @check_in
+        raise ArgumentError, "End date cannot be before start date"
```

Comment:
It's helpful to include the bad argument in the message when raising an `ArgumentError`:

```suggestion
        raise ArgumentError, "End date (#{end_date}) cannot be before start date (#{start_date})"
```